pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/SegmentBuildTimeLeaseExtender.java

```diff
@@ -46,25 +46,25 @@
   // Retransmit lease request 10% before lease expires.
   private static final int REPEAT_REQUEST_PERIOD_SEC = (EXTRA_TIME_SECONDS * 9 / 10);
   private static Logger LOGGER = LoggerFactory.getLogger(SegmentBuildTimeLeaseExtender.class);
-  private static final Map<String, SegmentBuildTimeLeaseExtender> INSTANCE_TO_LEASE_EXTENDER = new HashMap<>(1);
+  private static final Map<String, SegmentBuildTimeLeaseExtender> TABLE_TO_LEASE_EXTENDER = new HashMap<>(1);
```

Comment:
Can you change `new HashMap<>(1) to new HashMap<>()`? I think that we specifically built the hashmap with the size 1 because we only kept 1 lease extender per server while we now keep 1 lease extender per table.

Reply:
Make this `ConcurrentHashMap` and remove the `synchronized` on line 60. `HashMap` is not thread-safe on line 57

Reply:
synchronized in line 60 is ok. we check and then insert if it is not there. Alternatively,  we can use `computeIfAbsent` to check and insert atomically.

Reply:
Use `ConcurrentHashMap` and remove `synchronized` for `create()` may potential to create multiple lease extender since `create()` is not atomic. I think maybe it's better to keep the `HashMap` and add `synchronized` to the `getLeaseExtender()` also.